test/inference-engines/sprinkler.test.ts

```diff
@@ -1,87 +1,23 @@
-  expect(infer({ GRASS_WET: ['F'] }).toFixed(4)).toBe('0.7664')
-}
-
-const infersGiveRainTrue = (engine: IInferenceEngine) => {
-  const { infer } = engine
-  engine.setEvidence({ RAIN: ['T'] })
-
-  expect(infer({ SPRINKLER: ['T'] }).toFixed(4)).toBe('0.0100')
-  expect(infer({ SPRINKLER: ['F'] }).toFixed(4)).toBe('0.9900')
-  expect(infer({ GRASS_WET: ['T'] }).toFixed(4)).toBe('0.8019')
-  expect(infer({ GRASS_WET: ['F'] }).toFixed(4)).toBe('0.1981')
-}
-
-const infersGiveRainFalse = (engine: IInferenceEngine) => {
-  const { infer } = engine
-  engine.setEvidence({ RAIN: ['F'] })
-
-  expect(infer({ SPRINKLER: ['T'] }).toFixed(4)).toBe('0.4000')
-  expect(infer({ SPRINKLER: ['F'] }).toFixed(4)).toBe('0.6000')
-  expect(infer({ GRASS_WET: ['T'] }).toFixed(4)).toBe('0.3600')
-  expect(infer({ GRASS_WET: ['F'] }).toFixed(4)).toBe('0.6400')
-}
-
-const infersGiveGrassWetTrue = (engine: IInferenceEngine) => {
-  const { infer } = engine
-  engine.setEvidence({ GRASS_WET: ['T'] })
-
-  expect(infer({ RAIN: ['T'] }).toFixed(4)).toBe('0.3577')
-  expect(infer({ RAIN: ['F'] }).toFixed(4)).toBe('0.6423')
-  expect(infer({ SPRINKLER: ['T'] }).toFixed(4)).toBe('0.6467')
-  expect(infer({ SPRINKLER: ['F'] }).toFixed(4)).toBe('0.3533')
-}
-
-const infersGiveGrassWetFalse = (engine: IInferenceEngine) => {
-  const { infer } = engine
-  engine.setEvidence({ GRASS_WET: ['F'] })
-
-  expect(infer({ RAIN: ['T'] }).toFixed(4)).toBe('0.0718')
-  expect(infer({ RAIN: ['F'] }).toFixed(4)).toBe('0.9282')
-  expect(infer({ SPRINKLER: ['T'] }).toFixed(4)).toBe('0.0580')
-  expect(infer({ SPRINKLER: ['F'] }).toFixed(4)).toBe('0.9420')
-}
-
-const tests: { [key: string]: (engine: IInferenceEngine) => void } = {
-  'infers give Sprinkler True': infersGiveSprinklerTrue,
-  'infers give Sprinkler False': infersGiveSprinklerFalse,
-  'infers give Rain True': infersGiveRainTrue,
-  'infers give Rain False': infersGiveRainFalse,
-  'infers give Grass Wet True': infersGiveGrassWetTrue,
-  'infers give Grass Wet False': infersGiveGrassWetFalse,
-}
-
-describe('infers', () => {
-  describe('sprinkler network', () => {
-    const testNames = Object.keys(tests)
-    const engine = new InferenceEngine(network)
-
-    for (const testName of testNames) {
-      const method = tests[testName]
-
-      it(`${testName}`, () => method(engine))
-    }
-  })
-})
+import { runTests } from './helpers'
+
+// These gold standard values were computed indepdendently in R.
+const GOLD_STANDARD: [string[], number][] = [
+  [['T', 'T', 'T'], 1.98000000000000E-03],
+  [['F', 'T', 'T'], 2.88000000000000E-01],
+  [['T', 'F', 'T'], 1.58400000000000E-01],
+  [['F', 'F', 'T'], 0.00000000000000E+00],
+  [['T', 'T', 'F'], 2.00000000000000E-05],
+  [['F', 'T', 'F'], 3.20000000000000E-02],
+  [['T', 'F', 'F'], 3.96000000000000E-02],
+  [['F', 'F', 'F'], 4.80000000000000E-01],
+]
+
+const names: string[] = ['RAIN', 'SPRINKLER', 'GRASS_WET']
+const testValues: string[][][] = [
+  [['T'], ['F'], ['T', 'F']],
```

Comment:
Are all of these lines supposed to be the same?

Reply:
Yup.   All the variables are boolean ("T" or "F").   This will test all possible combinations of assignments ( zero, one or two values) to each variable